sdk/core/azure-core/src/main/java/com/azure/core/util/logging/LoggingEventBuilder.java

```diff
@@ -280,45 +284,51 @@ public RuntimeException log(RuntimeException runtimeException) {
         return runtimeException;
     }
 
+    /**
+     * Creates the JSON representation for the logging event.
+     *
+     * @param message the message to log.
+     * @param throwable {@link Throwable} for the message.
+     * @return JSON representation for the logging event.
+     * @throws UncheckedIOException If an I/O error occurs.
+     */
     private String getMessageWithContext(String message, Throwable throwable) {
         if (message == null) {
             message = "";
         }
 
-        StringBuilder sb = new StringBuilder(
-            20 + (context == null ? 0 : context.size()) * 20 + message.length() + globalContextCached.length());
-        // message must be first for log parsing tooling to work, key also works as a
-        // marker for Azure SDK logs so we'll write it even if there is no message
-        sb.append(AZURE_SDK_LOG_MESSAGE_JSON_START);
-        JSON_STRING_ENCODER.quoteAsString(message, sb);
-        sb.append('"');
-
-        if (throwable != null) {
-            sb.append(",\"exception\":");
-
-            // todo (alzimmer): Is adding '"exception": null' useful?
-            String exceptionMessage = throwable.getMessage();
-            if (exceptionMessage != null) {
-                sb.append('"');
-                JSON_STRING_ENCODER.quoteAsString(exceptionMessage, sb);
-                sb.append('"');
-            } else {
-                sb.append("null");
+        try (AccessibleByteArrayOutputStream outputStream = new AccessibleByteArrayOutputStream();
+            JsonWriter jsonWriter = JsonProviders.createWriter(outputStream)) {
+            jsonWriter.writeStartObject().writeStringField("az.sdk.message", message);
+
+            if (throwable != null) {
+                // todo (alzimmer): Is adding '"exception": null' useful?
+                String exceptionMessage = throwable.getMessage();
+                if (exceptionMessage != null) {
+                    jsonWriter.writeStringField("exception", exceptionMessage);
+                } else {
+                    jsonWriter.writeNullField("exception");
+                }
             }
-        }
 
-        if (hasGlobalContext) {
-            sb.append(',').append(globalContextCached);
-        }
+            if (hasGlobalContext) {
+                jsonWriter.flush();
```

Comment:
is this necessary?

Reply:
It can be if the JsonWriter implementation has a buffer that allows it to not need to flush contents to the backing store each write operation. But this could be removed by using a global context Map instead of a pre-serialized string.